### PR TITLE
Dispose BinaryReader, BinaryWriter and MemoryStream in AxHost.State

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Looks like that some disposables should be disposed in this class.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/3478)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/AxHost.State.cs [hunk 1]
```diff
@@ -118,7 +118,8 @@ protected State(SerializationInfo info, StreamingContext context)
                             if (dat != null)
                             {
                                 PropertyBagBinary = new PropertyBagStream();
-                                PropertyBagBinary.Read(new MemoryStream(dat));
+                                using MemoryStream datMemoryStream = new MemoryStream(dat);
```
Comment (hughbe) [inline src/System.Windows.Forms/src/System/Windows/Forms/AxHost.State.cs:121]: in L104 we also create but don't dispose a MemoryStream

File: src/System.Windows.Forms/src/System/Windows/Forms/AxHost.State.cs [hunk 2]
```diff
@@ -369,9 +370,9 @@ void ISerializable.GetObjectData(SerializationInfo si, StreamingContext context)
                 {
                     try
                     {
-                        stream = new MemoryStream();
-                        PropertyBagBinary.Write(stream);
-                        si.AddValue(nameof(PropertyBagBinary), stream.ToArray());
+                        using MemoryStream propertyBagBinaryStream = new MemoryStream();
```
Comment (hughbe) [inline src/System.Windows.Forms/src/System/Windows/Forms/AxHost.State.cs:373]: nit: var

File: src/System.Windows.Forms/src/System/Windows/Forms/AxHost.State.cs (hunk 1 above)
Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/AxHost.State.cs]: Good catch.

File: src/System.Windows.Forms/src/System/Windows/Forms/AxHost.State.cs (hunk 2 above)
Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/AxHost.State.cs]: I used var in all changed lines.

**Pull request merged**